source/dub/recipe/io.d

```diff
@@ -82,7 +82,9 @@ PackageRecipe parsePackageRecipe(string contents, string filename, string parent
 	{
 		try {
 			ret = parseConfigString!PackageRecipe(contents, filename, mode);
-			fixDependenciesNames(ret.name, ret);
+			// nothrow because otherwise parseJson might write into a partly
+			// initialized PackageRecipe, if this method throws.
+			() nothrow { fixDependenciesNames(ret.name, ret); } ();
```

Comment:
Unneeded ?

Reply:
I didn't want to check if the old parseJson can work with an already initialized PackageRecipe, so I added nothrow here, so it's not possible that the catch block is entered with an already deserialized json and possibly including all the array values twice.

We can also just mark the fixDependenciesNames as nothrow, but it was kinda bad to comment why that templated function needs to be forced nothrow, so I thought it might be better to just make the call site explicitly document and prove that it needs nothrow.

Reply:
But you already marked it nothrow and IMO that's better.
The only reason it wasn't nothrow in the first place was the AA iteration.